sanityv3/schemas/objects/textTeaser.tsx

```diff
@@ -0,0 +1,142 @@
+]
+
+const blockConfig = {
+  h1: false,
+  h2: false,
+  h3: false,
+  h4: false,
+  internalLink: false,
+  externalLink: false,
+  attachment: false,
+  lists: true,
+}
+
+const blockContentType = configureBlockContent({ ...blockConfig })
+
+export type TextTeaser = {
+  _type: 'textTeaser'
+  title?: PortableTextBlock[]
+  text?: PortableTextBlock[]
+  action?: (LinkSelector | DownloadableFile | DownloadableImage)[]
+  titlePosition?: string
+  theme?: ThemeSelectorValue
+}
+
+export default {
+  name: 'textTeaser',
+  title: 'Text Teaser',
+  type: 'object',
+  localize: true,
+  fieldsets: [
+    {
+      name: 'link',
+      title: 'Link',
+      description: 'Select either an internal link or external URL.',
+    },
+    {
+      name: 'design',
+      title: 'Design options',
+    },
+  ],
+  fields: [
+    {
+      name: 'title',
+      type: 'array',
+      components: {
+        input: CompactBlockEditor,
+      },
+      of: [titleContentType],
+      validation: (Rule: Rule) => Rule.required(),
+    },
+    {
+      name: 'text',
+      title: 'Text content',
+      type: 'array',
+      of: [blockContentType],
+      validation: (Rule: Rule) =>
+        Rule.custom((value: PortableTextBlock[]) => {
+          return validateCharCounterEditor(value, 600)
+        }).warning(),
+    },
+
+    {
+      name: 'action',
+      title: 'Link/action',
+      description: 'Select the link or downloadable file for the teaser',
+      type: 'array',
+      of: [
+        { type: 'linkSelector', title: 'Link' },
+        { type: 'downloadableImage', title: 'Downloadable image' },
+        { type: 'downloadableFile', title: 'Downloadable file' },
+      ],
+      validation: (Rule: Rule) => Rule.max(1).error('Only one action is permitted'),
+    },
+    {
+      name: 'titlePosition',
+      title: 'Title position',
+      description: 'Select which side of the teaser the title should be displayed at on larger screens.',
+      type: 'string',
+      fieldset: 'design',
+      components: {
+        input: function ({ onChange, value }: { onChange: any; value: string }) {
+          return (
+            <RadioIconSelector
+              name="imageAlignmentSelector"
+              options={titleAlignmentOptions}
+              defaultValue="left"
+              currentValue={value}
+              onChange={onChange}
+            />
+          )
+        },
+      },
+    },
+    {
+      title: 'Theme',
+      description: 'Pick a colour combination for the background and title highlight. Default is white.',
+      name: 'theme',
+      type: 'themeList',
+      fieldset: 'design',
+    },
+  ],
+  preview: {
+    select: {
+      title: 'title',
+      text: 'text',
+    },
+    prepare({ title, text }: { title: PortableTextBlock[]; text: PortableTextBlock[] }) {
+      const plainTitle = blocksToText(title)
+      return {
+        title: plainTitle || 'Missing teaser title',
+        subtitle: blocksToText(text) || 'Mising teaser text',
```

Comment:
Would recommentd prefixing the string 'Text Teaser | ' before adding subtitle content.. helps editors visualizing what section it is before going into the component. Also nice to add a media if you can find an interesting icon